### ISSUE TITLE
Detecting a global window object

### ISSUE THREAD

**Issue by chinedufn** [rustwasm/wasm-bindgen#896]
#847 made it so that we detect a `Window` global by getting calling `Function ()()` and taking that returned `this` and seeing if it is `instanceof Window`.

One thing that I'll sometimes do that isn't a best practice but is nice when you only care about running tests quickly and are less concerned about "being in a browser" is attach a `Window` to the `global` in Node.js

Example: https://github.com/chinedufn/percy/blob/master/tests/jsdom/tests/test-utils.js#L11-L21

This used to work, but after #847 it no longer does.

---

Any chance that instead of trying to see if the global context is an `instanceof Window` we can just see if there is a `window` property on the global context?

Maybe by using `.has_own_property("window")` or something?

Again.. not a best practice by any means.. but certainly a nice to have when you want a quick test and aren't concerned about running in a real browser all that much.

**Comment by chinedufn** [comment]
Yeah so I tried testing in wasm-bindgen-test (geckodriver) for the first time and I had no idea that the drivers spun up that quicklyy... so this is no longer an issue for me..

**Comment by fitzgen** [comment]
You can also use `js_sys::global()` and `JsCast` or `js_sys::Reflect` to do this kind of feature detection
